sale_triple_discount/models/sale_order_line.py

```diff
@@ -11,22 +11,53 @@
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
-    @api.depends('discount2', 'discount3')
+    def _get_final_discount(self):
+        self.ensure_one()
+        if self.discounting_type == "additive":
+            return self._additive_discount()
+        elif self.discounting_type == "multiplicative":
+            return self._multiplicative_discount()
```

Comment:
It should return 0 (zero) if no discount_type configured, I think.
I got None when running some tests and then it breaks the later computation

Reply:
I made the field `discount_type` required. As it has a default it will pick the correct value if not set in data

Reply:
@mpanarin  But should we not be able to have a sale_order_line with no discount applied to it ?

Reply:
@TDu we can set the discount to 0